framework/validators/UniqueValidator.php

```diff
@@ -234,7 +244,20 @@ private function prepareConditions($targetAttribute, $model, $attribute)
             $conditions = [$targetAttribute => $model->$attribute];
         }
 
-        return $conditions;
+        if (!$model instanceof ActiveRecord) {
+            return $conditions;
+        }
+
+        // Add table prefix for column.
+        $targetClass = $this->getTargetClass($model);
+        $tableName = $targetClass::tableName();
+        $conditionsWithTableName = [];
```

Comment:
@vladis84, @samdark - OK, I missed that. But I still think that `$conditionsWithTableName` variable can be just something like `$prefixedConditions`. In this context it's clear what it means. Also it's shorter and more readable IMO. The same applies to other related variables such as `$columnNameWithPrefix` (can be just `$prefixedColumn`). What do you think?